File: 0663-equal-tree-partition/0663-equal-tree-partition.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var checkEqualTree = function(root) {
    var seen = [];
    var dfs = function(node) {
        if(!node) return 0;
        const subSum = node.val + dfs(node.left) + dfs(node.right);
        seen.push(subSum);
        return subSum;
    }
    const total = dfs(root);
    seen.pop();
    for(const s of seen) {
        if(2*s == total) return true;
    }
    return false;
};

File: 0663-equal-tree-partition/0663-equal-tree-partition.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public bool CheckEqualTree(TreeNode root) {
        List<int> seen = new List<int>();
        int dfs(TreeNode node) {
            if(node == null) return 0;
            int subSum = node.val + dfs(node.left) + dfs(node.right);
            seen.Add(subSum);
            return subSum;
        }
        int total = dfs(root);
        seen.RemoveAt(seen.Count-1);
        foreach(int s in seen)
            if(2 * s == total) return true;
        return false;
    }
}